overlord/configstate/configmgr.go

```diff
@@ -41,9 +43,28 @@ func MockConfigcoreRun(f func(sysconfig.Device, configcore.RunTransaction) error
 	}
 }
 
+// Get home directories from system-params file.
+// Home directories are returned as a ',' separated string
+func homeDirsSysParams(rootDir string) error {
+
+	ssp, err := sysparams.Open(rootDir)
+	if err != nil {
+		dirs.SnapHomeDirs = ""
+		return err
+	}
+	dirs.SnapHomeDirs = ssp.Homedirs
+	return nil
+}
+
 func Init(st *state.State, hookManager *hookstate.HookManager) error {
 	delayedCrossMgrInit()
 
+	// Retrieve the list of home directories
+	err := homeDirsSysParams(dirs.GlobalRootDir)
```

Comment:
here we don't need to read the file, we can access the value from the option in the state. With config.NewTransaction and GetMaybe